backend/common/src/main/java/ai/verta/modeldb/common/event/FutureEventDAO.java

```diff
@@ -0,0 +1,111 @@
+package ai.verta.modeldb.common.event;
+
+import ai.verta.modeldb.common.config.Config;
+import ai.verta.modeldb.common.exceptions.ModelDBException;
+import ai.verta.modeldb.common.futures.FutureJdbi;
+import ai.verta.modeldb.common.futures.InternalFuture;
+import com.google.gson.JsonObject;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Executor;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class FutureEventDAO {
+  private static final Logger LOGGER = LogManager.getLogger(FutureEventDAO.class);
+  private final Executor executor;
+  private final FutureJdbi jdbi;
+  private final Config config;
+  private final String serviceType;
+
+  public FutureEventDAO(Executor executor, FutureJdbi jdbi, Config config, String serviceType) {
+    this.executor = executor;
+    this.jdbi = jdbi;
+    this.config = config;
+    this.serviceType = serviceType;
+  }
+
+  public Boolean addLocalEventWithBlocking(
+      String resourceType, String eventType, long workspaceId, JsonObject eventMetadata) {
+    return addLocalEvent(resourceType, eventType, workspaceId, eventMetadata).get();
+  }
+
+  public InternalFuture<Boolean> addLocalEventWithAsync(
+      String resourceType, String eventType, long workspaceId, JsonObject eventMetadata) {
+    return addLocalEvent(resourceType, eventType, workspaceId, eventMetadata);
+  }
+
+  private InternalFuture<Boolean> addLocalEvent(
+      String resourceType, String eventType, long workspaceId, JsonObject eventMetadata) {
+    if (!config.isEvent_system_enabled()) {
+      LOGGER.info("Event system is not enabled");
+      return InternalFuture.completedInternalFuture(false);
+    }
+
+    if (workspaceId == 0) {
+      LOGGER.info("workspaceId not found while logging event");
+      return InternalFuture.failedStage(new ModelDBException("Workspace id should not be empty"));
+    }
+
+    eventMetadata.addProperty("service", serviceType);
+    eventMetadata.addProperty("resource_type", resourceType);
+    eventMetadata.addProperty("logged_time", new Date().getTime());
+
+    return jdbi.withHandle(
+            handle ->
+                handle
+                    .createUpdate(
+                        "insert into event (event_uuid, event_type, workspace_id, event_metadata) values (:event_uuid, :event_type, :workspace_id, :event_metadata) ")
+                    .bind("event_uuid", UUID.randomUUID().toString())
+                    .bind("event_type", eventType)
+                    .bind("workspace_id", workspaceId)
+                    .bind("event_metadata", eventMetadata.toString())
+                    .execute())
+        .thenApply(
+            insertedRowCount -> {
+              LOGGER.debug("Event added successfully");
```

Comment:
What if the row count is 0? do you want to still do the successful log in that case?